presto-main/src/main/java/com/facebook/presto/execution/ClusterSizeMonitor.java

```diff
@@ -159,6 +185,15 @@ public boolean hasRequiredCoordinators()
         return currentCoordinatorCount >= coordinatorMinCountActive;
     }
 
+    /**
+     * @return true when the current coordinator sidecars count in a cluster is greater or equals to
+     * minimum coordinator sidecars count for a given Coordinator.
+     */
+    public boolean hasRequiredCoordinatorSidecars()
```

Comment:
Let's simplify this and just make it 1.  We can just make the logic to wait for a single sidecar, and if there are multiple, error out.